pvlib/iotools/nasa_power.py

```diff
@@ -0,0 +1,150 @@
+"""Function for reading and retrieving data from NASA POWER."""
+
+import pandas as pd
+import requests
+
+
+URL = 'https://power.larc.nasa.gov/api/temporal/hourly/point'
+
+VARIABLE_MAP = {
+    'ALLSKY_SFC_SW_DWN': 'ghi',
+    'ALLSKY_SFC_SW_DIFF': 'dhi',
+    'ALLSKY_SFC_SW_DNI': 'dni',
+    'CLRSKY_SFC_SW_DWN': 'ghi_clear',
+    'T2M': 'temp_air_2m',
+    'WS2M': 'wind_speed_2m',
+    'WS10M': 'wind_speed_10m',
+}
+
+
+def get_nasa_power(latitude, longitude, start, end, parameters,
+                   time_standard='utc', community='re', elevation=None,
+                   wind_height=None, wind_surface=None, map_variables=True):
+    """
+    Retrieve irradiance and weather data from NASA POWER.
+
+    A general description of NASA POWER is given in [1]_ and the API is
+    described in [2]_. A detailed list of the available parameters can be
+    found in [3]_.
+
+    Parameters
+    ----------
+    latitude: float
+        In decimal degrees, north is positive (ISO 19115).
+    longitude: float
+        In decimal degrees, east is positive (ISO 19115).
+    start: datetime like
+        First timestamp of the requested period. If a timezone is not
+        specified, UTC is assumed.
+    end: datetime like
+        Last timestamp of the requested period. If a timezone is not
+        specified, UTC is assumed.
+    parameters: str, list
```

Comment:
To make this easier to use, I suggest adding a list of default parameters. These could include GHI, DHI, DNI, temperature, and wind speed.

Reply:
Note, there are only 36 parameters available with API with an hourly frequency.